src/bin/pg_resetwal/pg_resetwal.c

```diff
@@ -488,6 +504,15 @@ main(int argc, char *argv[])
 		exit(1);
 	}
 
+#ifdef PERCONA_EXT
+	/*
+	 * As we always write empty record here we can do it in unencrypted mode.
+	 * We initialize writes here because new WAL key may be created in some cases,
+	 * and we want it only when we are ready to perform acutal writes.
```

Comment:
I struggle to understand this comment tbh. I guess, although we won't encrypt anything, we still need a new key and to write via xlog_smgr->seg_write so when we start the server, TDE can identify that this segment doesn't need decryption.

What do you mean by "new WAL key may be created in some cases,  and we want it only when we are ready to perform acutal writes."?

Reply:
Terrible sentence for sure. I wanted to point why writes initialized on this line and not together with `TDEXLogSmgrInit` above. We want `TDEXLogSmgrInitWrite` to be invoked only when we checked all arguments and other stuff because this function will create new key **if** encryption was enabled before. If it wasn't no new key will be added.

Rewrote this section.